pkg/odo/cli/cli.go

```diff
@@ -139,6 +139,9 @@ func odoRootCmd(ctx context.Context, name, fullName string, testClientset client
 
 	commonflags.AddOutputFlag()
 	commonflags.AddPlatformFlag(ctx)
+	if os.Args[1] == dev.RecommendedCommandName {
```

Comment:
This will panic if len(Args) < 2

Reply:
Also, do we want to treat this flag as a global one? For the moment, it will be used only for `odo dev`

Reply:
> Also, do we want to treat this flag as a global one? For the moment, it will be used only for odo dev

Don't wnat to, but this was the only way to make this flag experimental. Checking experimental mode requires `ctx`, which is not available in the local command (`NewCmdDev`).

Reply:
Could you pass the ctx to the `NewCmdDev` function, as it is done for `NewCmdPreference`, `NewCmdSet`, etc?